controllers/cart.js

```diff
@@ -0,0 +1,149 @@
+const Cart = require("../models/Cart");
+var ObjectId = require("mongodb").ObjectId;
+//const User=require("../models/user")
+const User = require("../models/user");
+const bcryptjs = require("bcryptjs");
+
+/*
+exports.addToCart = async (req, res, next) => {
+  try {
+    const { userId, productItems, quantity, price } = req.body;
+    const cartData = await Cart.create({
+      userId,
+      productItems,
+      quantity,
+      price,
+      total: price * quantity,
+    });
+
+    return res.status(200).json({
+      success: true,
+      message: "Data saved successfully.",
+      data: cartData,
+    });
+  } catch (error) {
+    console.log(error);
+    return res.status(500).json({
+      success: false,
+      message:
+        "We are having some error while completing your request. Please try again after some time.",
+      error: error,
+    });
+  }
+};
+*/
+exports.getCart = async (req, res) => {
+  try {
+    const cart = await Cart.findOne(req.params.user)

```

Comment:
As discussed you are supposed to fetch userid from token. User will not send explicitly in API